zerver/views/registration.py

```diff
@@ -756,7 +762,14 @@ def create_realm(request: HttpRequest, creation_key: Optional[str] = None) -> Ht
             if key_record is not None:
                 key_record.delete()
             new_realm_send_confirm_url = reverse("new_realm_send_confirm")
-            query = urlencode({"email": email})
+            query = urlencode(
+                {
+                    "email": email,
+                    "realm_name": realm_name,
+                    "realm_type": realm_type,
+                    "realm_subdomain": realm_subdomain,
+                }
+            )
```

Comment:
Are we fine with adding this parameters to the url or we want to pass these as separate parameters to the request?

Reply:
Redirects don't support passing parameters another way, so I think this is the best option if we're not going to use cookies or something.